cli/azd/internal/vsrpc/util_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package vsrpc
+
+import (
+	"embed"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+//go:embed all:testdata/samples/*
```

Comment:
This stuff is starting to get splayed across the tree (for both our acceptance tests, where we actually run the `dotnet` toolchain against them) and the unit level tests in aspire (where it's helpful to have these artifacts on disk even without running them (and instead mock the behavior of .NET).

I want to centralize all this stuff in a `internal/samples` package (that has all the embeds exported as well as `coptSample` and friends and sort of make this common technology our tests can leverage.

Can do that before landing this or after depending on how everyone feels - I wanted to just unblock myself on the plane and focus on writing these tests vs a larger samples refactoring.

Interested in your thoughts here, @weikanglim, since you had been writing tests that need similar technolog?  I sort of find value in having a few different shapes of aspire apps and manifests for my testing.

Reply:
I haven't given this thought the amount of duration it deserves, but I really like the locality between the fixtures for the test and the test itself is nice, so my initial inkling is to lean away from having a central location.

But having a library that simplifies some of this work does make a lot of sense. I draw heavy inspiration from some projects I see out there, where there's an "acceptance test library" that is heavily used to avoid the extra boilerplate, but each go package and test package is largely independent from one another.